2010-04-15  Michael Tautschnig  <mt@debian.org>

	* simple example: Added class GRUB_PC that installs and uses grub-pc instead
		of grub, following the suggestions of Jean Spirat <jeanspirat@squirk.org>.
    Thanks Waldemar Brodkorb <fai@waldemar-brodkorb.de> for more info and
    debugging.
	* Makefile: Make sure that all example scripts are executable
Index: trunk/Makefile
===================================================================
--- trunk.orig/Makefile
+++ trunk/Makefile	
@@ -47,6 +47,7 @@
 	install -p -m644 pixmaps/*.gif $(DESTDIR)/usr/share/fai/pixmaps
 	perl -pi -e 's/FAIVERSIONSTRING/$(VERSIONSTRING)/' $(DESTDIR)/usr/sbin/fai
 	cp -a examples $(DOCDIR)
+	chmod -R a+x $(DOCDIR)/examples/simple/scripts/
 	cp -a utils $(DOCDIR)/examples
 	find $(DOCDIR) -name .svn | xargs rm -rf
 
Index: trunk/examples/simple/package_config/DEFAULT
===================================================================
--- trunk.orig/examples/simple/package_config/DEFAULT
+++ trunk/examples/simple/package_config/DEFAULT	
@@ -17,10 +17,10 @@
 dhcp3-client
 
 PACKAGES aptitude GRUB
-grub lilo-
+grub lilo- grub-pc-
 
 PACKAGES aptitude GRUB_PC
 grub-pc grub- lilo-
 
 PACKAGES aptitude LILO
-lilo grub-
+lilo grub- grub-pc-
Index: trunk/examples/simple/scripts/GRUB_PC/10-setup
===================================================================
--- trunk.orig/examples/simple/scripts/GRUB_PC/10-setup
+++ trunk/examples/simple/scripts/GRUB_PC/10-setup	
@@ -2,8 +2,32 @@
 
 error=0 ; trap "error=$((error|1))" ERR
 
-$ROOTCMD grub-mkdevicemap -n -m /boot/grub/device.map
-$ROOTCMD grub-mkconfig -o /boot/grub/grub.cfg
-$ROOTCMD grub-install --no-floppy "(hd0)"
+set -a
+
+# during softupdate use this file
+[ -r $LOGDIR/disk_var.sh ] && . $LOGDIR/disk_var.sh
+
+[ -z "$BOOT_DEVICE" ]    && exit 701
+[ -z "$BOOT_PARTITION" ] && exit 702
+
+# if class NOMBR is defined, write boot loader into root partition, not into mbr
+ifclass NOMBR && BOOT_DEVICE=$BOOT_PARTITION
+
+$ROOTCMD grub-mkimage --output=/boot/grub/core.img biosdisk part_msdos ext2 lvm raid
+$ROOTCMD update-grub
+
+for device in $BOOT_DEVICE; do
+  if echo $device | grep -q "^/dev/mapper/" ; then
+    grub_dev=$(echo $device | sed 's#^/dev/mapper/##')
+  else
+    grub_dev=$(device2grub $device)
+    if [ -z "$grub_dev" ] ; then
+      grub_dev=$(echo $device | sed 's#^/dev/##')
+    fi
+  fi
+  $ROOTCMD /usr/sbin/grub-install --no-floppy --modules="biosdisk part_msdos ext2 lvm raid" "($grub_dev)"
+  $ROOTCMD /usr/sbin/grub-setup   "($grub_dev)"
+done
 
 exit $error
+
Index: trunk/examples/simple/class/50-host-classes
===================================================================
--- trunk.orig/examples/simple/class/50-host-classes
+++ trunk/examples/simple/class/50-host-classes	
@@ -20,5 +20,4 @@
 	echo "FAIBASE DHCPC" ;;
 esac
 
-(ifclass I386 || ifclass AMD64) && echo GRUB 
 exit 0
Index: trunk/examples/simple/class/51-grub
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ trunk/examples/simple/class/51-grub	
@@ -0,0 +1,4 @@
+#! /bin/bash
+
+{ ifclass I386 || ifclass AMD64; } && ! ifclass GRUB_PC && echo GRUB
+exit 0
